### PR TITLE
[plugin.video.popcorntv] 1.0.17

### PR DESCRIPTION
**Pull request by nightflyer73**
### Description
Updated page parsing because of upstream HTML changes.
This addon now requires youtube addon. Updates will only be available for krypton and higher.

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.popcorntv/addon.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.video.popcorntv"
+       name="PopcornTv"
+       version="1.0.17"
+       provider-name="Nightflyer">
+  <requires>
+    <import addon="xbmc.python" version="2.1.0"/>
```
Comment (enen92) [inline plugin.video.popcorntv/addon.xml]: 2.25.0 please

File: plugin.video.popcorntv/addon.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    <import addon="plugin.video.youtube" version="5.1.16"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource"
+            library="default.py">
+        <provides>video</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+    <language>it</language>
+    <platform>all</platform>
+    <summary lang="it">La web TV On-Demand, legale e gratuita. Cinema, Musica, Sport e tanto di più</summary>
```
Comment (enen92) [inline plugin.video.popcorntv/addon.xml]: it_IT

File: plugin.video.popcorntv/addon.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <description lang="it">La Over the Top TV, completamente gratuita, in streaming on demand, di film, video musicali, avvenimenti sportivi, e numerosi altri contenuti di una Library in continuo aggiornamento.</description>
+    <summary lang="en">The legal and free On-Demand web TV. Movies, Music, Sports and much more</summary>
```
Comment (enen92) [inline plugin.video.popcorntv/addon.xml]: en_GB please (new default since krypton)

File: plugin.video.popcorntv/default.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    xbmcplugin.addSortMethod(handle, xbmcplugin.SORT_METHOD_LABEL)
+    xbmcplugin.endOfDirectory(handle=handle, succeeded=True)
+    
+def show_video_files(url):
+    xbmc.log("Subcategory URL: " + url)
+    popcorntv = PopcornTV()
+    page = popcorntv.getVideoBySubCategories(url)
+    
+    for video in page["videoList"]:
+        liStyle=xbmcgui.ListItem(video["title"], thumbnailImage=video["thumb"])
```
Comment (enen92) [inline plugin.video.popcorntv/default.py]: thumbnailImage kwarg is deprecated please use liStyle.setArt({"thumb":xxxx, "icon": xxx, "fanart": xxxx, etc})

File: plugin.video.popcorntv/default.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        liStyle.setInfo('video', {"Plot": video["plot"]})
+        addLinkItem({"mode": "video", "url": video["url"]}, liStyle)
+              
+    if page["prevPageUrl"] is not None:
+        liStyle=xbmcgui.ListItem("< Prev Page", iconImage = "DefaultFolder.png")
```
Comment (enen92) [inline plugin.video.popcorntv/default.py]: would be nice to not have this hardcoded but in a strings file (note this is not a blocking issue but just a suggestion)

File: plugin.video.popcorntv/default.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        addDirectoryItem({"mode": "list", "url": page["prevPageUrl"]}, liStyle)
+        
+    if page["nextPageUrl"] is not None:
+        liStyle=xbmcgui.ListItem("> Next Page", iconImage = "DefaultFolder.png")
```
Comment (enen92) [inline plugin.video.popcorntv/default.py]: iconimage deprecated (setArt as mentioned above)

File: plugin.video.popcorntv/addon.xml (hunk 2 above)
Comment (musuruan) [reply plugin.video.popcorntv/addon.xml]: @enen92 What about the language tag? "it" is fine there or have I to change it there too.

Comment (nightflyer73) [reply plugin.video.popcorntv/addon.xml]: @enen92 What about the language tag? "it" is fine there or have I to change it there too.

Comment (enen92) [reply plugin.video.popcorntv/addon.xml]: Language tag still uses the old format :) keep it as it

**Pull request merged**